modules/caas/backend/src/main/java/io/cattle/platform/agent/server/resource/impl/AgentResourcesMonitor.java

```diff
@@ -251,7 +249,10 @@ protected void setIpAddresses(Map<String, Host> hosts, Agent agent, AgentResourc
                 Map<String, Object> labels = CollectionUtils.toMap(data.get(HostConstants.FIELD_LABELS));
                 labels.putAll(CollectionUtils.toMap(data.get(HostConstants.FIELD_CREATE_LABELS)));
                 data.put(HostConstants.FIELD_LABELS, labels);
-
+                Cluster cluster = objectManager.loadResource(Cluster.class, agent.getClusterId());
+                if (cluster != null && StringUtils.equals(DataAccessor.fieldString(cluster, ClusterConstants.FIELD_ORCHESTRATION), ClusterConstants.ORCH_KUBERNETES) && !cluster.getEmbedded()) {
```

Comment:
protip :)  If you are comparing string to a constant, then do  `CONSTANT.equals(someString)`  That will ensure you never get a NullPointerException